app/forms/search_form.rb

```diff
@@ -17,6 +20,15 @@ class SearchForm
   attribute :triage_status, :string
   attribute :year_groups, array: true
 
+  attr_accessor :session, :request_path
+
+  def initialize(params = {}, session:, request_path:)
+    @session = session
+    @request_path = request_path
+    super(params)
```

Comment:
```suggestion
  def initialize(params = {})
    super(params)
```

I think `ActiveModel::Model` will handle the assignments for us.

Reply:
`ActiveModel::Model` does indeed handle assignments, but only for defined attributes. I chose to configure the arguments to differentiate between `params` and other collaborator objects such as `session` and `request_path`. If we want to go ahead with the suggested approach, we'll have to define  `session` and `request_path` as attributes.

Then we'd have to change the usage so that we have something like:

`SearchForm.new(**params, session:, request_path:)` or `SearchForm.new(params.merge(session:, request_path:))`.

What do you think?

Reply:
I think that's fine, we already do a similar approach in our other forms (for example here https://github.com/nhsuk/manage-vaccinations-in-schools/blob/d6ee775bc090b817209d3b9aa1b80ff03fd6860f/app/forms/batch_form.rb#L7) where the `batch` doesn't come from the query parameters, but it's still passed in like the attributes that do.